distributed/worker.py

```diff
@@ -766,6 +766,12 @@ def __init__(
 
         self.memory_limit = parse_memory_limit(memory_limit, self.nthreads)
 
+        if "thread_auto_interrupt" in kwargs:
+            self.interruptor = kwargs.pop("thread_auto_interrupt")
```

Comment:
imo, if we want to support this in the constructor, we should add it explicitly and add a doc string to it. I would be fine with supporting this merely via config, though.